agent/osv_agent.py

```diff
@@ -30,36 +30,38 @@ def process(self, message: m.Message) -> None:
         """Process messages of type v3.asset.file and scan dependencies against vulnerabilities.
         Once the scan is completed, it emits messages of type : `v3.report.vulnerability`
         """
-        self._osv_file_handler: osv_file_handler.OSVFileHandler
         logger.info("processing message of selector : %s", message.selector)
         content = message.data.get("content")
         path = message.data.get("path")
+        if path is None:
+            logger.error("Can't process empty path.")
+            return
         if content is None or content == b"":
             logger.warning("Message file content is empty.")
             return
-        self._osv_file_handler = osv_file_handler.OSVFileHandler(
-            content=content, path=path
-        )
 
-        self._run_osv(content)
+        self._run_osv(content, path)
```

Comment:
path then content.

Reply:
You should only pase the last part of the file. Like if the path is /tmp/a.json only pass a.json.

Also you should write the file is a special location to avoid file overwrite and be vulnerable to something bad.

Reply:
it is already the case, I ll change the order of args.